examples/bls12-381-hash/src/bls12-381-hash.rs

```diff
@@ -325,18 +327,869 @@ fn g2_clear_cofactor(p: G2) -> G2 {
 }
 
 // https://www.ietf.org/archive/id/draft-irtf-cfrg-hash-to-curve-13.html#section-3
-pub fn g2_hash_to_curve(msg: &ByteSeq, dst: &ByteSeq) -> G2 {
+pub fn g2_hash_to_curve_svdw(msg: &ByteSeq, dst: &ByteSeq) -> G2 {
     let u = fp2_hash_to_field(msg, dst, 2);
-    let q0 = g2_map_to_curve(u[0]);
-    let q1 = g2_map_to_curve(u[1]);
+    let q0 = g2_map_to_curve_svdw(u[0]);
+    let q1 = g2_map_to_curve_svdw(u[1]);
     let r = g2add(q0, q1);
     let p = g2_clear_cofactor(r);
     p
 }
 
-pub fn g2_encode_to_curve(msg: &ByteSeq, dst: &ByteSeq) -> G2 {
+pub fn g2_encode_to_curve_svdw(msg: &ByteSeq, dst: &ByteSeq) -> G2 {
     let u = fp2_hash_to_field(msg, dst, 1);
-    let q = g2_map_to_curve(u[0]);
+    let q = g2_map_to_curve_svdw(u[0]);
+    let p = g2_clear_cofactor(q);
+    p
+}
+
+/* Simplified SWU for AB == 0 method
+*
+*
+*/
+
+// https://www.ietf.org/archive/id/draft-irtf-cfrg-hash-to-curve-13.html#section-6.6.2
+fn g1_simple_swu_iso(u: Fp) -> (Fp, Fp) {
+    let z = Fp::from_literal(11u128);
+    let a = Fp::from_byte_seq_be(&ArrFp::to_be_bytes(&ArrFp(secret_array!(
```

Comment:
You could move these constants out to have a cleaner function body.
Something like this
```rust
const G1_A: ArrFp = ArrFp(secret_array!(
    U64,
    [
        0x00144698a3b8e943u64,
        0x3d693a02c96d4982u64,
        0xb0ea985383ee66a8u64,
        0xd8e8981aefd881acu64,
        0x98936f8da0e0f97fu64,
        0x5cf428082d584c1du64
    ]
));

// ...

    let a = Fp::from_byte_seq_be(&G1_A.to_be_bytes());
```

Same for all the other constants in functions.